kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/ClusterEventTracking.java

```diff
@@ -421,10 +421,6 @@ protected void writeBatchEvents(Collection<Event> events)
 
                 // process the insert
                 if (cachingEnabled) {
-                    conn = sqlService().borrowConnection();
```

Comment:
This looks good but it only affects those using terracotta which is very few.

Reply:
This part of the fix looks good.

Reply:
I thought many was using terracotta. Maybe there is no performance gain of the ClusterEventTracking?

I think the ClusterEventTracking should be split up in 2 parts. One part that is saving the events to the shared cache (should happen immediately) and one part for persisting the events. Persisting the events could be done by anyone in the cluster, like once every minute. But this is out of topic for the pull request.